### PR TITLE
Added VideoControls class with play/pause, rewind, & > < buttons

### PR DESCRIPTION
**Pull request by benjaminwp18**

### CODE REVIEW

File: gui/app.py [hunk 1]
```diff
@@ -20,32 +20,111 @@ class VideoThread(QThread):
...
     def __init__(self, filenames):
         super().__init__()
-        self._run_flag = True
+        self._thread_running_flag = True
+        self._video_playing_flag = True
         self._filenames = filenames
         self._captures = []
+        self._rewind = False
+
+    def __emit_frames(self):
```
Comment (ryankarpuszka) [inline gui/app.py:29]: I would only use one underscore at the beginning of this name

File: gui/app.py [hunk 2]
```diff
@@ -20,32 +20,111 @@ class VideoThread(QThread):
...
+        self.prev_frame_button.setText("<")
+        self.prev_frame_button.clicked.connect(self.thread.prev_frame)
+        self.horizontal_layout.addWidget(self.prev_frame_button)
+
+        self.next_frame_button = QPushButton(self)
+        self.next_frame_button.setText(">")
+        self.next_frame_button.clicked.connect(self.thread.next_frame)
+        self.horizontal_layout.addWidget(self.next_frame_button)
+
+        self.setLayout(self.horizontal_layout)
```
Comment (ryankarpuszka) [inline gui/app.py:127]: Could you try moving this to a separate file so we don't clutter this file up too much?

File: gui/app.py [hunk 3]
```diff
@@ -20,32 +20,111 @@ class VideoThread(QThread):
...
-            self.msleep(int(1000 / 30))
+        # Run the play/pausable video 
+        while self._thread_running_flag:
+            # Send frames while the video is playing
+            while self._thread_running_flag and self._video_playing_flag:
+                self.__emit_frames()
+                self.msleep(int(1000 / 30))
+            
+            # Take a NOP while its paused
+            while self._thread_running_flag and not self._video_playing_flag:
```
Comment (ryankarpuszka) [inline gui/app.py:62]: You could get rid of the nested while loops to simplify things a bit.
```suggestion
            if self._video_playing_flag:
                self._emit_frames()
            
            self.msleep(...)
```

**Review by ryankarpuszka — CHANGES_REQUESTED** [review]
This is all working pretty well. Just some suggestions I had.

**Comment by ryankarpuszka** [comment]
Resolves #12

**Comment by benjaminwp18** [comment]
I've made the changes you suggested; I can't believe I wrote that loop spaghetti.

File: gui/app.py [hunk 4]
```diff
@@ -1,14 +1,14 @@
 from PyQt5 import QtGui
-from PyQt5.QtWidgets import QWidget, QApplication, QLabel, QVBoxLayout, QComboBox, QTabWidget
+from PyQt5.QtWidgets import QHBoxLayout, QPushButton, QWidget, QApplication, QLabel, QComboBox, QVBoxLayout, QTabWidget
 from PyQt5.QtGui import QPixmap
-import cv2
 from PyQt5.QtCore import pyqtSignal, pyqtSlot, Qt, QThread
+import cv2
 import numpy as np
 
+from video_controls_widget import VideoControlsWidget
```
Comment (ryankarpuszka) [inline gui/app.py:8]:
```suggestion
from gui.video_controls_widget import VideoControlsWidget
```

**Review by ryankarpuszka — CHANGES_REQUESTED** [review]
There's still calls to `__emit_frames` instead of `_emit_frames`.

**Review by ryankarpuszka — APPROVED** [review]
Looks good!

**Pull request merged**